app/src/main/java/com/quarkworks/apartmentgroceries/grocery/AddGroceryItemActivity.java

```diff
@@ -106,4 +162,139 @@ public Void then(Task<Void> task) throws Exception {
             return null;
         }
     };
+
+    private class ImageAdapter extends BaseAdapter {
+        private Context context;
+        private ArrayList<byte[]> photoList;
+
+        public ImageAdapter(Context context, ArrayList<byte[]> photoList) {
+            if (photoList == null) photoList = new ArrayList<>();
+            this.context = context;
+            this.photoList = photoList;
+
+        }
+
+        public int getCount() {
+            return photoList.size();
+        }
+
+        public Object getItem(int position) {
+            return null;
+        }
+
+        public long getItemId(int position) {
+            return 0;
+        }
+
+        public View getView(final int position, View convertView, ViewGroup parent) {
+            ImageView imageView;
+            View rootView = View.inflate(context, R.layout.add_grocery_grid_view_item, null);
+            imageView = (ImageView) rootView.findViewById(R.id.add_grocery_grid_view_image_view_id);
+
+            byte[] photoBytes = photoList.get(position);
+            Bitmap bitmap = BitmapFactory.decodeByteArray(photoBytes, 0, photoBytes.length);
+            imageView.setImageBitmap(bitmap);
+
+            imageView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    if (position == photoList.size() - 1) {
```

Comment:
This would be best put in another file since this file has gotten fairly large now.